fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingByRecords.java

```diff
@@ -77,7 +76,8 @@ public class IndexingByRecords extends IndexingBase {
 
     IndexingByRecords(@Nonnull IndexingCommon common, @Nonnull OnlineIndexer.IndexingPolicy policy) {
         super(common, policy);
-        this.recordsRange = computeRecordsRange();
+        final TupleRange range = common.computeRecordsRange();
+        this.recordsRange = range == null ? TupleRange.ALL : range;
```

Comment:
Rather than retaining both `IndexingByRecords` and `IndexingMultiTargetByRecords` until the range based APIs can be implemented on both (or removed), would it instead make sense to try and isolate the logic that truly is different between single- and multi-index situations, and then delete one of the classes? From what I can tell, the differences are:

1. When computing the records range, all record types from all indexes need to be included so as to cover all needed records
2. When indexing, all index maintainers need to be updated
3. When updating index states, each index needs to be updated

There are maybe a few other things, but this PR seems to also touch code related to picking ranges to scan, which I think should actually be common between the indexing methods

Reply:
Sorry, but I'm not sure that I understand this comment. Can you clarify?

Reply:
(after re-reading) 
Yes, I do not wish to retain both modules but to (eventually) delete `IndexingByRecords`, or just leave it with minimal functionality for API backward compatibility - if we decide to keep it. 
The multi-target module also has a slightly different flow, which is simpler and more compatible with the other new modules. It's probably a good idea to sharing code when it's possible and easy - like `computeRecordsRange` here..